### DISCUSSION TITLE
Carousel buttons overflowing full width page layout

### DISCUSSION THREAD

**Discussion by ridhwaans** [shadcn-ui/ui#5483] [General]
![image](https://github.com/user-attachments/assets/b24b9d83-25a6-47bb-8db2-2b749750f9f2)
I have a few carousel components in a nextjs page with root layout.tsx. There is also a Nav component floating left
I use the Carousel component however I think the absolute positioning of the left and right arrow turn buttons are causing a horizontal scrollbar on what is supposed to be a full width page

Below is the additional styling I have
```
 <Carousel
        opts={{
          align: 'start',
          loop: true,
        }}
        className="grid relative w-full"
      >
        <CarouselContent className="-ml-1">
  ```

Thanks in advance
https://ui.shadcn.com/docs/components/carousel

**Comment by toniuyt123** [comment]
Did you manage to solve this?

**Reply by ridhwaans** [reply]
yes, add absolute position css to `CarouselPrevious` and `CarouselNext` to be within the carousel container
```javascript
      ...
      </CarouselContent>
        <div className="absolute top-1/2 left-2 flex items-center justify-center">
          <CarouselPrevious className="relative left-0 translate-x-0 hover:translate-x-0 hover:bg-primary/90" />
        </div>
        <div className="absolute top-1/2 right-2 flex items-center justify-center">
          <CarouselNext className="relative right-0 translate-x-0 hover:translate-x-0 hover:bg-primary/90" />
        </div>
     </Carousel>
  ```
